### DISCUSSION TITLE
Prevent iframes re-rendering on drop?

### DISCUSSION THREAD

**Discussion by KrisCoulson** [clauderic/dnd-kit#502] [Q&A]
Is there a way to prevent re-rendering an iframe on dragEnd? Since dnd-kit uses the state to update the position this causes a re-render on dragEnd. If the draggable item contains an iframe it is reloaded every time the item is repositioned. If you move an item from position 3 to position 1 it will reload the iframe in positions 2 and 3. I have been kinda bouncing around if handling this is possible with dnd-kit. Any thoughts on a way to make this work?

https://codesandbox.io/s/magical-dijkstra-3xmfu?file=/src/App.js

**Comment by clauderic** [comment]
You can prevent re-renders using `React.memo` and stable keys: https://reactjs.org/docs/react-api.html#reactmemo

However, the use-case you are trying to build is not possible to achieve because of a limitation in the `iframe` element when it is moved in the DOM, refer to https://stackoverflow.com/questions/8318264/how-to-move-an-iframe-in-the-dom-without-losing-its-state

You could get around this problem by using a `video` element instead.

**Reply by mikehw** [reply]
Would it be possible to maintain the order of elements in the DOM but apply a transform after on dragEnd to make the elements appear in the document the way the user placed them, and account for that transform when subsequent drag transforms are applied?
